widget/embedded/src/components/SwapDetails/SwapDetails.tsx

```diff
@@ -0,0 +1,399 @@
+                    numberToString(outputAmount, MIN_DECIMALS, AMOUNT_DECIMALS)
+                  ) *
+                    (getUsdPrice(
+                      lastStep.toBlockchain,
+                      lastStep.toSymbol,
+                      lastStep.toSymbolAddress,
+                      tokens
+                    ) || 0)
+                ),
+                MIN_DECIMALS,
+                AMOUNT_DECIMALS
+              ),
+            }}
+            token={{
+              displayName: steps[numberOfSteps - 1].to.token.displayName,
+              image: steps[numberOfSteps - 1].to.token.image,
+            }}
+            chain={{ image: steps[numberOfSteps - 1].to.chain.image }}
+            percentageChange={numberToString(
+              percentageChange,
+              PERCENT_DECIMALS,
+              PERCENT_DECIMALS
+            )}
+            warningLevel={getPriceImpactWarningLevel(percentageChange)}
+          />
+        </div>
+        <div className="title-steps">
+          <Typography variant="title" size="small">
+            {i18n.t('Swaps steps')}
+          </Typography>
+        </div>
+        <Divider size={8} />
+        <StepsList shouldRetry={shouldRetry} ref={listRef}>
+          {steps.map((step, index) => {
+            const key = index;
+            const state = getStepState(swap.steps[index]);
+            const isFocused =
+              state === 'error' ||
+              state === 'in-progress' ||
+              state === 'warning' ||
+              (state === 'completed' && index === steps.length - 1);
+            return (
+              <StepDetails
+                key={key}
+                step={step}
+                type="route-progress"
+                ref={listRef}
+                state={state}
+                hasSeparator={index !== 0}
+                tabIndex={key}
+                isFocused={isFocused}
+              />
+            );
+          })}
+        </StepsList>
+        {shouldRetry && !showCompletedModal && (
+          <StyledButton
+            variant="contained"
+            type="primary"
+            size="large"
+            onClick={() => {
+              retry(swap);
+              setTimeout(() => {
+                navigate(navigationRoutes.home);
+              }, 0);
+            }}>
+            {i18n.t('Try again')}
+          </StyledButton>
+        )}
+      </Container>
+      <SwapDetailsModal
+        state={modalState}
+        onClose={() => setModalState(null)}
+        onCancel={onCancel}
+        onDelete={onDelete}
+        message={stepMessage.detailedMessage.content}
+        currentStepWallet={currentStepWallet}
+      />
+      <SwapDetailsCompleteModal
+        open={!!showCompletedModal}
+        diagnosisUrl={diagnosisUrl}
+        onClose={() => setShowCompletedModal(null)}
+        status={showCompletedModal}
+        priceValue={numberToString(outputAmount, MIN_DECIMALS, AMOUNT_DECIMALS)}
+        usdValue={numberToString(
```

Comment:
Could you keep the value of `usdValue` and `description` in a separate variable to be more readable.

Reply:
fixed.